File: model/src/main/java/ar/edu/itba/paw/model/Category.java
package ar.edu.itba.paw.model;

import java.util.Objects;

public enum Category {
    ITALIAN("category.italian.name"),
    AMERICAN("category.american.name"),
    SPANISH("category.spanish.name"),
    PIZZA("category.pizza.name"),
    STEAKHOUSE("category.steakhouse.name"),
    BAR("category.bar.name"),
    CHINESE("category.chinese.name"),
    VIET_THAI("category.viet_thai.name"),
    BURGERS("category.burgers.name"),
    JAPANESE("category.japanese.name"),
    SUSHI("category.sushi.name"),
    VEGGIE("category.veggie.name"),
    INDIAN("category.indian.name"),
    EMPANADAS("category.empanadas.name"),
    BOULANGERIE("category.boulangerie.name"),
    COFFEE("category.coffee.name"),
    BREWERY("category.brewery.name"),
    PERUVIAN("category.peruvian.name"),
    NIKKEI("category.nikkei.name"),
    ARABIAN("category.arabian.name"),
    MEXICAN("category.mexican.name"),
    GELATO("category.gelato.name");

    private final String message;

    Category(String message) {
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    public long getId() {
        return ordinal();
    }

    public String getIdString() {
        return String.valueOf(getId());
    }

    public static Category getById(long id) {
        for (Category c : Category.values()) {
            if (c.ordinal() == id) {
                return c;
            }
        }
        return null;
    }

    public static Category getByMessage(String message) {
        for (Category c : Category.values()) {
            if (Objects.equals(c.message, message)) {
                return c;
            }
        }
        return null;
    }

}

File: dine-out-app/src/types/enums/Category.ts
export class Category {
    static readonly ITALIAN = new Category('ITALIAN', 'Category.italian')
    static readonly AMERICAN = new Category('AMERICAN', 'Category.american')
    static readonly SPANISH = new Category('SPANISH', 'Category.spanish')
    static readonly PIZZA = new Category('PIZZA', 'Category.pizza')
    static readonly STEAKHOUSE = new Category('STEAKHOUSE', 'Category.steakhouse')
    static readonly BAR = new Category('BAR', 'Category.bar')
    static readonly CHINESE = new Category('CHINESE', 'Category.chinese')
    static readonly VIET_THAI = new Category('VIET_THAI', 'Category.viet_thai')
    static readonly BURGERS = new Category('BURGERS', 'Category.burgers')
    static readonly JAPANESE = new Category('JAPANESE', 'Category.japanese')
    static readonly SUSHI = new Category('SUSHI', 'Category.sushi')
    static readonly VEGGIE = new Category('VEGGIE', 'Category.veggie')
    static readonly INDIAN = new Category('INDIAN', 'Category.indian')
    static readonly EMPANADAS = new Category('EMPANADAS', 'Category.empanadas')
    static readonly BOULANGERIE = new Category('BOULANGERIE', 'Category.boulangerie')
    static readonly COFFEE = new Category('COFFEE', 'Category.coffee')
    static readonly BREWERY = new Category('BREWERY', 'Category.brewery')
    static readonly PERUVIAN = new Category('PERUVIAN', 'Category.peruvian')
    static readonly NIKKEI = new Category('NIKKEI', 'Category.nikkei')
    static readonly ARABIAN = new Category('ARABIAN', 'Category.arabian')
    static readonly MEXICAN = new Category('MEXICAN', 'Category.mexican')
    static readonly GELATO = new Category('GELATO', 'Category.gelato')

    private constructor (private readonly name: string, public readonly description: string) {

    }

    toString (): string {
        return this.description
    }
}